scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -1156,4 +1158,46 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
     )
     new MultiMapSideInput[K, V](o)
   }
+
+  /**
+   * Convert this SCollection to a SideInput, mapping key-value pairs of each window to a
+   * `Map[key, value]`, to be used with [[SCollection.withSideInputs]]. It is required that each
+   * key of the input be associated with a single value.
+   *
+   * Currently, the resulting map is required to fit into memory. This is preferable to
+   * [[asMapSideInput]] if that's the case.
+   */
+  def asMapSingletonSideInput(implicit koder: Coder[K], voder: Coder[V]): SideInput[Map[K, V]] =
+    self
+      .transform(
+        _.groupByKey
```

Comment:
wondering if we could have a combine step here instead?

Reply:
Yeah was being lazy. Figured <100MB it might not make a huge difference. I'll take a look.

Reply:
Actually I'm not sure how much performance gain we get for the cost of readability. `Iterable` probably has O(n) append & concat so we'll have to use mutable/vector/whatever & cast later, adds a fair amount of cruft.

Reply:
In this particular case the combiner would be only `Map[K, V]` and adding to an immutable map is effectively constant so I think it would be a lot faster. Also with this iirc step fusion might not be guaranteed

Reply:
The merge fns `(Map[K, V], (K, V)) => Map[K, V]` & `(Map[K, V], Map[K, V]) => Map[K, V])` will still have to check keys with >1 values though.